### PR TITLE
Rename UCP names and Context Type with TAE(Tanzu Application Engine)

### PR DESCRIPTION
**Pull request by prkalle**
### What this PR does / why we need it
This PR renames all the instances of UCP names and ContextType with TAE(Tanzu Application Engine)

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

Unit tests and CI passed

### Release note

```release-note
Rename UCP names and Context Type with TAE(Tanzu Application Engine)
```

### CODE REVIEW

File: tae/tae.go [hunk 1]
```diff
@@ -16,14 +16,14 @@ import (
...
-	"github.com/vmware-tanzu/tanzu-plugin-runtime/ucp/internal/kubeconfig"
+	"github.com/vmware-tanzu/tanzu-plugin-runtime/tae/internal/kubeconfig"
 )
 
 // keys to Context's AdditionalMetadata map
 const (
-	OrgIDKey       = "ucpOrgID"
-	ProjectNameKey = "ucpProjectName"
-	SpaceNameKey   = "ucpSpaceName"
+	OrgIDKey       = "taeOrgID"
```
Comment (prkalle) [inline tae/tae.go:24]: @vuil I hope renaming "ucpOrgID" to "taeOrgID" should be ok. Similarly for `ucpProjectName` and `ucpSpaceName`

**Review by marckhouzam — APPROVED** [review]
LGTM

File: tae/tae.go (hunk 1 above)
Comment (vuil) [reply tae/tae.go:24]: yes, it would make sense for consistency to do so.

**Review by vuil — APPROVED** [review]
lgtm, thanks.

File: config/types/clientconfig_helper.go [hunk 2]
```diff
@@ -27,7 +27,8 @@ func IsValidTarget(target string, allowGlobal, allowUnknown bool) bool {
 		target == string(TargetK8s) ||
 		target == string(targetTMC) ||
 		target == string(TargetTMC) ||
-		target == string(TargetUCP) ||
+		target == string(targetTAE) ||
+		target == string(TargetTAE) ||
 		(allowGlobal && target == string(TargetGlobal)) ||
```
Comment (anujc25) [inline config/types/clientconfig_helper.go:32]: nit: `target == string(targetTAE)` is specified twice.

**Pull request merged**